app/lib/efile/ny/it213.rb

```diff
@@ -3,154 +3,292 @@ module Ny
         @value_access_tracker = value_access_tracker
         @lines = lines
         @filing_status = filing_status
         @direct_file_data = direct_file_data
+        @eligibility_lived_in_state = eligibility_lived_in_state
         @federal_dependent_child_count = federal_dependent_child_count
-        @federal_dependent_child_count_between_4_and_17 = federal_dependent_child_count_between_4_and_17
       end
 
       def calculate
-        # TODO: Do we need to consider Worksheet B?
-        # TODO: Only calculate Worksheet A if yes on line 2? Should only happen if they clicked the wrong federal button?
-        # TODO: Only run calculate if yes in line 1, and 3
-        set_line(:IT213_LINE_3, :calculate_line_3)
-        set_line(:IT213_LINE_4, -> { @federal_dependent_child_count})
-        set_line(:IT213_LINE_5, -> { @federal_dependent_child_count_between_4_and_17})
+        set_line(:IT213_LINE_1, -> { @eligibility_lived_in_state })
+        # If line 1 is false, stop, you do not qualify for the credit
+        if @eligibility_lived_in_state
+          set_line(:IT213_LINE_2, -> { @direct_file_data.fed_ctc_claimed })
+          set_line(:IT213_LINE_3, :calculate_line_3)
+          # If lines 2 and 3 are false, stop, you do not qualify for this credit
+          if @lines[:IT213_LINE_2].value || @lines[:IT213_LINE_3].value
+            set_line(:IT213_LINE_4, -> { @federal_dependent_child_count})
+            set_line(:IT213_LINE_5, -> { 0 }) # TODO: double check that people with children without SSN/ITIN are out of scope
+            # If line 2 is true, you must complete Worksheet A and B before you continue with line 6.
+            # If line 2 is false, skip lines 6 through 8, and enter 0 on line 9; continue with line 10.
+            if @lines[:IT213_LINE_2].value
+              calculate_worksheets
+            else
+              set_line(:IT213_LINE_9, -> { 0 })
+            end
+            # If line 3 is false, skip lines 10 through 13, and enter the amount from line 9 on line 14.
+            # All others continue with line 10.
+            if @lines[:IT213_LINE_3].value
+              set_line(:IT213_LINE_10, :calculate_line_10)
+              set_line(:IT213_LINE_11, :calculate_line_11)
+              set_line(:IT213_LINE_12, :calculate_line_12)
+              set_line(:IT213_LINE_13, :calculate_line_13)
+              set_line(:IT213_LINE_14, :calculate_line_14)
+            else
+              set_line(:IT213_LINE_14, -> { @lines[:IT213_LINE_9].value })
+            end
+            # TODO: If we wanted to support spouse filing separate, lines 15 and 16 could have share of credit for spouse
+            set_line(:IT213_LINE_15, -> { 0 })
+            set_line(:IT213_LINE_16, -> { 0 })
+          else
+            offboard
+            return
+          end
+        else
+          offboard
+        end
+      end
```

Comment:
hmm yeah i wanted to mimic the flow of the instructions as much as possible but i don't think the way i did it is quite right. i think i'll actually switch the two checks that might result in offboarding from "if" to "unless" so the first branch is offboard and otherwise you continue. that matches the instructions closest.

Reply:
nevermind, tried it and it just got messier. gonna go with your suggestion plus a comment 👍